src/pybind/mgr/prometheus/module.py

```diff
@@ -8,6 +8,8 @@
 import threading
 import time
 import enum
+import gc
```

Comment:
Instead of bringing Python's explicit `gc` into play (which can be a risky guest), we could use Python's `weakref` to avoid increasing the ref counting on the Python side. For example, that `healthcheck` data structure could be a `weakref.WeakValueDictionary` object. 

I thought that the `functools` cache helpers already used weak refs, but I was wrong. It's the non-standard `cachetools` [library the one that uses it](https://github.com/tkem/cachetools/blob/master/src/cachetools/_cachedmethod.py) (for key values as it uses it for function memoization).

Reply:
Invoking a `gc.collect()` in Python triggers a complete scan of all in-memory data, including deeply nested dicts/large collections. If we're not keeping complex data structures with circular references (where weakrefs are usually the best approach), there's no benefit in invoking `gc.collect()`. Ref-counting is enough to deallocate resources. Is this based on a real benchmarking?

Reply:
I can remove the gc.collect() now after changing the refcounts in the Cache itself. According to the massif output, the memory that the mgr api no longer leaks that memory